mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala

```diff
@@ -39,3 +39,66 @@ class Tokenizer extends UnaryTransformer[String, Seq[String], Tokenizer] {
 
   override protected def outputDataType: DataType = new ArrayType(StringType, false)
 }
+
+
+/**
+ * :: AlphaComponent ::
+ * A regex based tokenizer that extracts tokens using a regex.
+ * Optional additional parameters include enabling lowercase stabdarization, a minimum character
+ * size for tokens as well as an array of stop words to remove from the results.
+ */
+@AlphaComponent
+class RegexTokenizer extends UnaryTransformer[String, Seq[String], RegexTokenizer] {
+
+  val lowerCase = new BooleanParam(this, 
+      "lowerCase", 
+      "enable case folding to lower case", 
+      Some(true))
+  def setLowercase(value: Boolean) = set(lowerCase, value)
+  def getLowercase: Boolean = get(lowerCase)
+
+  val minLength = new IntParam(this, 
+      "minLength", 
+      "minimum token length (excluded)", 
+      Some(0))
+  def setMinLength(value: Int) = set(minLength, value)
+  def getMinLength: Int = get(minLength)
+
+  val regEx = new Param(this, 
+      "regEx", 
+      "RegEx used for tokenizing", 
+      Some("\\p{L}+|[^\\p{L}\\s]+".r))
+  def setRegex(value: scala.util.matching.Regex) = set(regEx, value)
+  def getRegex: scala.util.matching.Regex = get(regEx)
+
+  val stopWords = new Param(this, 
+      "stopWords", 
+      "array of tokens to filter from results", 
+      Some(Array[String]()))
+  def setStopWords(value: Array[String]) = set(stopWords, value)
+  def getStopWords: Array[String] = get(stopWords)
+
+
+  override protected def createTransformFunc(paramMap: ParamMap): String => Seq[String] = { x =>
+
+    var string = x
+    if (paramMap(lowerCase)) {
+      string = string.toLowerCase
+    }
+    var tokens = (paramMap(regEx) findAllIn string).toList
```

Comment:
Do not use infix (https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide#SparkCodeStyleGuide-InfixMethods).